lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchDriverConfigurationProviderImpl.java

```diff
@@ -58,14 +63,23 @@ private AzureBatchDriverConfigurationProviderImpl(
       @Parameter(AzureBatchPoolId.class) final String azureBatchPoolId,
       @Parameter(AzureStorageAccountName.class) final String azureStorageAccountName,
       @Parameter(AzureStorageContainerName.class) final String azureStorageContainerName,
+      @Parameter(TcpPortSet.class) final Set<Integer> tcpPortSet,
+      final ContainerRegistryProvider containerRegistryProvider,
       final CommandBuilder commandBuilder) {
     this.jvmSlack = jvmSlack;
     this.azureBatchAccountUri = azureBatchAccountUri;
     this.azureBatchAccountName = azureBatchAccountName;
     this.azureBatchPoolId = azureBatchPoolId;
     this.azureStorageAccountName = azureStorageAccountName;
     this.azureStorageContainerName = azureStorageContainerName;
+    this.containerRegistryProvider = containerRegistryProvider;
+
     this.commandBuilder = commandBuilder;
+    this.tcpPortSet = new HashSet<String>(tcpPortSet.size());
+    for (int port: tcpPortSet) {
```

Comment:
Why are we converting between integer and string?

Reply:
ConfigurationModule.setMultiple only accepts a set of strings. That is why I had to convert from ints to strings

---
In reply to: [201088444](https://github.com/sharathmalladi/reef/pull/5#discussion_r201088444) [](ancestors = 201088444)

Reply:
Please add a comment to explain this.

---
In reply to: [201473909](https://github.com/sharathmalladi/reef/pull/5#discussion_r201473909) [](ancestors = 201473909,201088444)